### PR TITLE
Fixed build for latest rust master

### LINKED ISSUE #2708: Incoming clippy breakage
Looks like https://github.com/rust-lang/rust/pull/50204 changes the name of `span_approximate_suggestion` to `span_suggestion_with_applicability` and adds an [extra suggestion type parameter](https://github.com/rust-lang/rust/pull/50204/files#diff-2c49ea48a32b1e12546e0517e544ef58R60).

### PR DESCRIPTION
**Pull request by alexreg**
Fixes #2708.

### CODE REVIEW

**Comment by oli-obk** [comment]
Retriggering travis

**Comment by oli-obk** [comment]
Clippy goes into an infinite loop on some input. I probably won't get to a machine where I can run gdb for the next few days :/

**Comment by alexreg** [comment]
Tried to run the tests locally, but getting failures like this for clippy:

```
---- [run-pass] run-pass/associated-constant-ice.rs stdout ----
	thread '[run-pass] run-pass/associated-constant-ice.rs' panicked at 'failed to exec `"target/debug/clippy-driver" "tests/run-pass/associated-constant-ice.rs" "-L" "/Users/alex/Software/rust/src/tools/clippy/target/debug/test_build_base" "--target=x86_64-apple-darwin" "-C" "prefer-dynamic" "-o" "/Users/alex/Software/rust/src/tools/clippy/target/debug/test_build_base/associated-constant-ice.stage-id" "-L" "target/debug" "-L" "target/debug/deps" "-Dwarnings" "-L" "/Users/alex/Software/rust/src/tools/clippy/target/debug/test_build_base/associated-constant-ice.stage-id.aux"`: Os { code: 2, kind: NotFound, message: "No such file or directory" }', libcore/result.rs:945:5
```

**Comment by mati865** [comment]
I don't see hangs on the Linux but inline and docs checks are messed up:
<details>
<summary>Click me</summary>

```
diff --git a/tests/ui/inline_fn_without_body.stderr b/tests/ui/inline_fn_without_body.stderr
index 2b466b68..e69de29b 100644
--- a/tests/ui/inline_fn_without_body.stderr
+++ b/tests/ui/inline_fn_without_body.stderr
@@ -1,26 +0,0 @@
-error: use of `#[inline]` on trait method `default_inline` which has no body
- --> $DIR/inline_fn_without_body.rs:8:5
-  |
-8 |       #[inline]
-  |  _____-^^^^^^^^
-9 | |     fn default_inline();
-  | |____- help: remove
-  |
-  = note: `-D inline-fn-without-body` implied by `-D warnings`
-
-error: use of `#[inline]` on trait method `always_inline` which has no body
-  --> $DIR/inline_fn_without_body.rs:11:5
-   |
-11 |     #[inline(always)]fn always_inline();
-   |     ^^^^^^^^^^^^^^^^^ help: remove
-
-error: use of `#[inline]` on trait method `never_inline` which has no body
-  --> $DIR/inline_fn_without_body.rs:13:5
-   |
-13 |       #[inline(never)]
-   |  _____-^^^^^^^^^^^^^^^
-14 | |     fn never_inline();
-   | |____- help: remove
-
-error: aborting due to 3 previous errors
-
diff --git a/tests/ui/missing-doc.stderr b/tests/ui/missing-doc.stderr
index 54834f90..ae2be387 100644
--- a/tests/ui/missing-doc.stderr
+++ b/tests/ui/missing-doc.stderr
@@ -78,6 +78,16 @@ error: missing documentation for a function
 51 | fn foo3() {}
    | ^^^^^^^^^^^^
 
+error: use of `#[inline]` on trait method `foo` which has no body
+  --> $DIR/missing-doc.rs:56:5
+   |
+56 |       /// dox
+   |  _____-^^^^^^
+57 | |     fn foo(&self);
+   | |____- help: remove
+   |
+   = note: #[deny(inline_fn_without_body)] on by default
+
 error: missing documentation for a trait
   --> $DIR/missing-doc.rs:68:1
    |
@@ -264,5 +274,5 @@ error: missing documentation for a function
 191 |         fn also_undocumented2() {}
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-error: aborting due to 39 previous errors
+error: aborting due to 40 previous errors
```
</details>

File: clippy_lints/src/inline_fn_without_body.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
 
 fn check_attrs(cx: &LateContext, name: &Name, attrs: &[Attribute]) {
     for attr in attrs {
-        if attr.name().map_or(true, |n| n != "inline") {
+        if attr.name() == "inline" {
```
Comment (mati865) [inline clippy_lints/src/inline_fn_without_body.rs]: Change `==` to `!=`.
With that change all tests passed on my Linux box.

**Comment by oli-obk** [comment]
Whooo. Thanks everyone for the issue search and/or commits! I'll publish a new version later today

**Comment by alexreg** [comment]
Thanks for spotting that! Sorry about the sloppy edit.

**Pull request merged**